lib/foreman/logging.rb

```diff
@@ -0,0 +1,169 @@
+require 'logging'
+
+module Foreman
+  class LoggingImpl
+    private_class_method :new
+
+    attr_reader :config, :root_logger, :log_directory
+
+    def configure(options = {})
+      fail 'logging can be configured only once' if @configured
+      @configured = true
+      @log_directory = options.fetch(:log_directory, './log')
+      @config = YAML.load_file('config/logging.yaml')
+      @root_logger = @config[:loggers][:root]
+
+      configure_color_scheme
+      begin
+        FileUtils.mkdir_p @log_directory unless File.directory?(@log_directory)
+      rescue Errno::EACCES
+        # just notify insufficient privileges
+        # rubocop:disable Output
+        puts "Insufficient privileges for #{@log_directory}"
+      end
+      configure_root_logger(options)
+      configure_children_loggers
+
+      if @config[:console_inline]
+        ::Logging.logger.root.add_appenders(
+            ::Logging.appenders.stdout(:layout => build_layout(@root_logger[:pattern], @config[:colorize])))
+      end
+
+      # you can add specific files per logger easily like this
+      #   Logging.logger['sql'].appenders = Logging.appenders.file("#{Rails.env}_sql.log")
+    end
+
+    def add_logger(logger_name, logger_config)
+      logger          = ::Logging.logger[logger_name]
+      logger.level    = logger_config[:level] if logger_config.key?(:level)
+      logger.additive = logger_config[:enabled] if logger_config.key?(:enabled)
+      logger.trace    = @config[:log_trace]
+      enable_tailing(logger, logger_config) if logger_config.key?(:tail_command)
```

Comment:
This option is undocumented (no example and comment in the default config).

Speaking about tailing, would it be too insane to include a simple API to tail logs from the Rails application itself, so we can create a tab in the About box to display last N lines of logs? :-)